app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -143,6 +145,23 @@ protected void onNewIntent(Intent unsafeIntent) {
             pendingUrl = intent.getDataString();
         }
 
+        if (Intent.ACTION_SEND.equals(intent.getAction())) {
+            String dataString = intent.getStringExtra(Intent.EXTRA_TEXT);
+            if (!TextUtils.isEmpty(dataString)) {
+                boolean isUrl = UrlUtils.isUrl(dataString);
+                TelemetryWrapper.shareIntentEvent(isUrl);
+                if (isUrl) {
+                    // We can't update our fragment right now because we need to wait until the activity is
+                    // resumed. So just remember this URL and load it in onResumeFragments().
+                    pendingUrl = dataString;
+                } else {
+                    // We can't update our fragment right now because we need to wait until the activity is
+                    // resumed. So just remember this URL and load it in onResumeFragments().
+                    pendingUrl = UrlUtils.createSearchUrl(this, dataString);
```

Comment:
nit: Maybe short syntax is more compact / readable here:

pendingUrl = isUrl ? .. : ..;